tests/samples/ReservedWords.h

```diff
@@ -0,0 +1,26 @@
+typedef int match;
+
+struct object {
+  match yield;
+  int val;
+};
+
+typedef struct object object;
+typedef struct object type;
+
+union lazy {
+  object *instance;
+  match forSome;
+  struct {
+    char def;
+    type *super;
+  } implicit;
+};
+
+typedef union lazy lazy;
+
+type with(match sealed, int implicit, lazy forSome);
+
+typedef match def;
+typedef struct { def val; lazy finally; } finally;
+match implicit(finally type[12]);
```

Comment:
I've added an additional test, but without the `const` keyword since it is not supported. The fix is to the type translator is not super clean, so there is probably some additional cleanups to do to make sure we escape keywords as late as possible.